src/hooks/fileTabs.hooks.ts

```diff
@@ -74,11 +74,23 @@ const useFileTab = () => {
     }
   };
 
-  const close = (filePath: string, closeAll: boolean = false) => {
+  const close = (
+    filePath: string | null,
+    action: 'close' | 'closeAll' | 'closeOthers' = 'close',
+  ) => {
     let updatedTab: IFileTab;
 
-    if (closeAll) {
+    if (action === 'closeAll') {
       updatedTab = { items: [], active: null };
+    } else if (action === 'closeOthers' && filePath) {
+      // Close all tabs except the specified one
+      const updatedItems = fileTab.items.filter(
+        (item) => item.path === filePath,
+      );
+      updatedTab = {
```

Comment:
Mutable `updatedTab` unnecessarily increases code complexity.

It'd be much easier to `return` instead, and factor out `syncTabSettings` out of this function.

Also `syncTabSettings` gets exported along with `open` and `close`, even though it's semantically/architecturally a deeper concept. Probably it's worth it to check if there is a way to avoid returning `syncTabSettings` from this function, and incapsulate all the logic related to tab syncing here.